SdarotAPI/SdarotDriver.cs

```diff
@@ -58,48 +71,32 @@ public async Task<bool> Login(string username, string password)
 
     public async Task<IEnumerable<SeriesInformation>> SearchSeries(string searchQuery)
     {
-        var searchHtml = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.SearchUrl}{searchQuery}");
-        var doc = new HtmlDocument();
-        doc.LoadHtml(searchHtml);
+        var showsJson = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.AllShowsUrl}");
+        var shows = JsonSerializer.Deserialize<List<SeriesInformation>>(showsJson);
+
+        var relevantShows = shows?.Where(x =>
```

Comment:
Better add a null check and return `Enumerable.Empty` instead of ignoring this scenario

Reply:
Then the Count() is relevant again, I'll do this

Reply:
Why that makes the `Count()` relevant?

Reply:
No Count property on IEnumerable, only method. But it doesn't matter, we removed this part completely.

Reply:
Let me clarify. The `?` makes the `relevantShow` variable to be nullable. I suggest the following:
```suggestion
        if (shows is null)
        {
                return Enumerable.Empty<SeriesInformation>();
        }

        var relevantShows = shows.Where(x =>
```

Reply:
Answered accidently on the next comment (with removing the `ToList` statement).